infinispan/runtime/src/main/java/io/quarkiverse/langchain4j/infinispan/SchemaAndMarshallerProducer.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkiverse.langchain4j.infinispan;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Instance;
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Inject;
+
+import org.infinispan.protostream.FileDescriptorSource;
+import org.infinispan.protostream.MessageMarshaller;
+
+import io.quarkiverse.langchain4j.infinispan.runtime.InfinispanEmbeddingStoreConfig;
+import io.quarkiverse.langchain4j.infinispan.runtime.LanchainItemMarshaller;
+
+@ApplicationScoped
+public class SchemaAndMarshallerProducer {
+
+    public static final String LANCHAIN_ITEM = "LanchainItem";
+
+    private static final String PROTO = "syntax = \"proto2\";\n" + "\n" + "/**\n" + " * @Indexed\n" + " */\n"
```

Comment:
Some other embedding stores allow configuring the name of the fields (in this case you have hardcoded `id`, `floatVector` and `text`). Wondering if it's actually useful and we should have it here too, maybe not.

In any case, we should also have a piece of documentation that describes the requirements on the schema of the cache - for the cases where the cache already exists on startup.

Reply:
I did nothing on documentation yet, yes this must be explained.

Reply:
@jmartisk concerning the naming, we can adapt the naming but I don't see the benefit right now.